core/src/components/datetime/utils/parse.ts

```diff
@@ -106,3 +106,29 @@ export const parseDate = (val: string | undefined | null): any | undefined => {
     tzOffset,
   };
 };
+
+/**
+ * Parses an hour and returns if the value is in the morning (am) or afternoon (pm).
+ * @param hour The hour to format, should be 0-23
+ * @returns `pm` if the hour is greater than or equal to 12, `am` if less than 12.
+ */
+export const parseAmPm = (hour: number) => {
+  return hour >= 12 ? 'pm' : 'am';
+};
+
+/**
+ * Parses the locale's string representation of the day period (am/pm) for a given
+ * date string value.
+ *
+ * @param locale The locale to format the day period in.
+ * @param value The date string, in ISO format.
+ * @returns The localized day period (am/pm) representation of the given value.
+ */
+export const parseLocalizedAmPm = (locale: string, dateStringIso: string) => {
+  return new Intl.DateTimeFormat(locale, {
+    hour: 'numeric',
+    timeZone: 'UTC',
+  })
+    .formatToParts(new Date(dateStringIso))
```

Comment:
The exact date string here isn't really important and it seems a bit odd to have to pass in the correct date string every time.

Have you considered making this a function that accepts a locale and either `'am'` or `'pm'` and then returns the formatted value?

Something like this:

```typescript
const formatAMPM = (locale: string, ampm: 'am' | 'pm') => {
  // You could even cache these in the file itself 
  // so you don't have to create the same date objects every invocation
  const date = (ampm === 'am') ? someAMDateObject : somePMDateObject;

  if (supportsDayPeriodInIntlDateTime) {
    /// create formatted label
    return formattedAMPMLabel;
  }

   // fallback for browsers that don't support dayPeriod
  return ampm;
}
```

There are a few benefits to this approach:

1. You can continue to just pass `refParts.ampm` in `getFormattedTime`. 
2. You would no longer have to create throwaway dates in https://github.com/ionic-team/ionic-framework/pull/25389/files#diff-4a407530c60e3cf72bcc11acdd21c4803a94bf47ea81b99e757db1c93d2735b8R1705.
3. You would no longer need an explicit fallback after the code in point 2 because `formatAMPM` handles the fallback.